api/src/main/java/net/kyori/adventure/text/ComponentLike.java

```diff
@@ -33,6 +38,45 @@
  */
 @FunctionalInterface
 public interface ComponentLike {
+  /**
+   * Converts a list of {@link ComponentLike}s to a list of {@link Component}s.
+   *
+   * @param likes the component-likes
+   * @return the components
+   * @since 4.8.0
+   */
+  static @NonNull List<Component> asComponents(final @NonNull List<? extends ComponentLike> likes) {
+    return asComponents(likes, null);
+  }
+
+  /**
+   * Converts a list of {@link ComponentLike}s to a list of {@link Component}s.
+   *
+   * <p>Only components that match {@code filter} will be returned.</p>
+   *
+   * @param likes the component-likes
+   * @param filter the component filter
+   * @return the components
+   * @since 4.8.0
+   */
+  static @NonNull List<Component> asComponents(final @NonNull List<? extends ComponentLike> likes, final @Nullable Predicate<? super Component> filter) {
+    if(likes.isEmpty()) {
+      // We do not need to create a new list if the one we are copying is empty - we can
+      // simply just return our known-empty list instead.
+      return Collections.emptyList();
+    }
+    final int size = likes.size();
+    final List<Component> components = new ArrayList<>(size);
```

Comment:
Consider calling `components.trimToSize();` or whatever it is on `ArrayList`, since this might way over-allocate.